ui/widget/src/main/java/de/danoeh/antennapod/ui/widget/WidgetUpdater.java

```diff
@@ -190,8 +164,77 @@ public static void updateWidget(Context context, WidgetState widgetState) {
                 views.setInt(R.id.butPlay, "setVisibility", View.VISIBLE);
             }
 
-            int backgroundColor = prefs.getInt(PlayerWidget.KEY_WIDGET_COLOR + id, PlayerWidget.DEFAULT_COLOR);
-            views.setInt(R.id.widgetLayout, "setBackgroundColor", backgroundColor);
+            Bitmap icon;
+            int radius = context.getResources().getDimensionPixelSize(R.dimen.widget_inner_radius);
+            RequestOptions option1 = new RequestOptions()
+                    .dontAnimate()
+                    .transform(new FitCenter(), new FastBlurTransformation())
+                    .diskCacheStrategy(DiskCacheStrategy.AUTOMATIC);
+
+            RequestOptions option2 = new RequestOptions()
+                    .dontAnimate()
+                    .transform(new FitCenter(), new RoundedCorners(radius));
+
+            if (showCoverAsBcg) {
+                views.setViewVisibility(R.id.imgvCover, View.GONE);
+                views.setViewVisibility(R.id.imgvBackground, View.VISIBLE);
+                //views.setViewVisibility(R.id.imgvBlurOverlay, View.VISIBLE);
+
+                try {
+                    icon = Glide.with(context)
+                            .asBitmap()
+                            .load(widgetState.media.getImageLocation())
+                            .apply(option1)
+                            .submit()
+                            .get(5000, TimeUnit.MILLISECONDS);
+                    views.setImageViewBitmap(R.id.imgvBackground, icon);
+
+                } catch (Throwable tr1) {
+                    try {
+                        icon = Glide.with(context)
+                                .asBitmap()
+                                .load(ImageResourceUtils.getFallbackImageLocation(widgetState.media))
+                                .apply(option1)
+                                .submit()
+                                .get(5000, TimeUnit.MILLISECONDS);
```

Comment:
Setting it to such a long timeout will cause issues with the widget not reacting. Please use the timeout you had before